test/ret_web/controllers/api/credentials_controller_test.exs

```diff
@@ -0,0 +1,362 @@
+defmodule RetWeb.CredentialsControllerTest do
+  use RetWeb.ConnCase
+  import Ret.TestHelpers, only: [create_account: 2, put_auth_header_for_account: 2]
+  alias Ret.Api.Scopes
+
+  @endpoint RetWeb.Endpoint
+
+  setup %{conn: conn} do
+    {
+      :ok,
+      conn: conn,
+      account: create_account("nonadmin", false),
+      admin_account: create_account("admin", true),
+      list: credentials_path(conn, :index, []),
+      create: credentials_path(conn, :create)
+    }
+  end
+
+  defp valid_scopes() do
+    [Scopes.read_rooms(), Scopes.write_rooms()]
+  end
+
+  defp app_only_scopes() do
+    [Scopes.read_rooms(), Scopes.write_rooms(), Scopes.create_accounts()]
+  end
+
+  defp params(scopes, subject_type) do
+    [scopes: scopes, subject_type: subject_type]
+  end
+
+  test "Accounts must authenticate to access credentials API.", %{
+    account: account,
+    conn: conn,
+    list: list,
+    create: create
+  } do
+    conn
+    |> put_auth_header_for_account(account)
+    |> get(list)
+    |> json_response(200)
+
+    conn |> get(list) |> response(401)
+
+    conn
+    |> put_auth_header_for_account(account)
+    |> post(create, params(valid_scopes(), :account))
+    |> json_response(200)
+
+    conn |> post(create, params(valid_scopes(), :account)) |> response(401)
+  end
+
+  test "Account credentials can be created and listed", %{
+    account: account,
+    conn: conn,
+    list: list,
+    create: create
+  } do
+    assert conn
+           |> put_auth_header_for_account(account)
+           |> get(list)
+           |> json_response(200)
+           |> Map.get("credentials")
+           |> Enum.count() === 0
+
+    conn
+    |> put_auth_header_for_account(account)
+    |> post(create, params(valid_scopes(), :account))
+    |> json_response(200)
+
+    assert conn
+           |> put_auth_header_for_account(account)
+           |> get(list)
+           |> json_response(200)
+           |> Map.get("credentials")
+           |> Enum.count() === 1
+  end
+
+  # TODO: Should this be allowed?
+  test "Admins accounts can create credentials on behalf of other accounts", %{
+    account: account,
+    admin_account: admin_account,
+    conn: conn,
+    list: list,
+    create: create
+  } do
+    assert conn
+           |> put_auth_header_for_account(account)
+           |> get(list)
+           |> json_response(200)
+           |> Map.get("credentials")
+           |> Enum.count() === 0
```

Comment:
We do this enough in the tests that it should be a `get_credentials_count` helper.